diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt

```diff
@@ -301,6 +303,19 @@ fun ASTNode?.isAccessibleOutside(): Boolean =
             true
         }
 
+fun ASTNode.hasSuppress(warningName: String): Boolean {
+    return if (elementType == FILE) {
+        findAllNodesWithSpecificType(ANNOTATION_ENTRY).any {
+            it.text.contains("Suppress(\"$warningName\")")
```

Comment:
There could be spaces around parenthesis (`@Suppress ( "WARNING" )`), so you should separately check annotation name and arguments. Look at `psi` as `KtAnnotationEntry`.